Readme.md
Moodle Course Index (block)
=============================

Course Index block

About
=====
 * copyright  &copy; 2022-onwards G J Barnard.
 * author     G J Barnard - http://about.me/gjbarnard and http://moodle.org/user/profile.php?id=442195
 * license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later

Required version of Moodle
==========================
This version works with:

 - Moodle 4.0dev+ version 2022011800.00 (Build: 20220118) and above within the 4.0 branch.

Please ensure that your hardware and software complies with 'Requirements' in 'Installing Moodle' on:
'docs.moodle.org/40/en/Installing_Moodle'
respectively.

Free Software
=============
The course index block is 'free' software under the terms of the GNU GPLv3 License, please see 'COPYING.txt'.

It can be obtained for free from:
https://github.com/gjb2048/moodle-block_course_index

You have all the rights granted to you by the GPLv3 license.  If you are unsure about anything, then the
FAQ - http://www.gnu.org/licenses/gpl-faq.html - is a good place to look.

If you reuse any of the code then I kindly ask that you make reference to the theme.

If you make improvements or bug fixes then I would appreciate if you would send them back to me by forking from
https://github.com/gjb2048/moodle-block_course_index and doing a 'Pull Request' so that the rest of the
Moodle community benefits.

Installation
============
1. Follow [Moodle's instructions for installing plugins](http://docs.moodle.org/40/en/Installing_plugins#Installation).
2. Turn editing mode on.
3. Add the "Course Index" block wherever you like - but the content will only show on the pages the core code allows.
2. Turn editing mode off.

Version Information
===================
See Changes.md

Me
==
G J Barnard MSc. BSc(Hons)(Sndw). MBCS. CEng. CITP. PGCE.
Moodle profile: http://moodle.org/user/profile.php?id=442195.
Web profile   : http://about.me/gjbarnard


Changes.md
Version Information
===================
Version 4.0.0.2
---------------------------
  1. Fix 'addmyinstance' capability not needed.
  2. State where the block can be added.

Version 4.0.0.1
---------------------------
  1. First alpha version for Moodle 4.0.


db/access.php
<?php
// This file is part of Moodle - http://moodle.org/
//
// Moodle is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Moodle is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.

/**
 * Course Index block.
 *
 * @package    block_course_index
 * @copyright  &copy; 2022-onwards G J Barnard.
 * @author     G J Barnard - {@link http://moodle.org/user/profile.php?id=442195}
 * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
 */

defined('MOODLE_INTERNAL') || die();

$capabilities = array(
    'block/course_index:addinstance' => array(
        'captype' => 'read',
        'contextlevel' => CONTEXT_BLOCK,
        'archetypes' => array(
            'editingteacher' => CAP_ALLOW,
            'manager' => CAP_ALLOW
        ),
        'clonepermissionsfrom' => 'moodle/site:manageblocks'
    )
);


block_course_index.php
<?php
// This file is part of Moodle - http://moodle.org/
//
// Moodle is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Moodle is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.

/**
 * Course Index block.
 *
 * @package    block_course_index
 * @copyright  &copy; 2022-onwards G J Barnard.
 * @author     G J Barnard - {@link http://moodle.org/user/profile.php?id=442195}
 * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
 */
defined('MOODLE_INTERNAL') || die;

require_once($CFG->dirroot.'/course/lib.php');

/**
 * Course Index block class.
 */
class block_course_index extends block_base {

    /**
     * Initialise the block.
     */
    public function init() {
        $this->title = get_string('pluginname', 'block_course_index');
    }

    /**
     * Defines where the block can be added.
     *
     * @return array
     */
    public function applicable_formats() {
        return array('course' => true, 'mod' => true);
    }

    /**
     * States if the block has a configuration.
     *
     * @return boolean success.
     */
    public function has_config() {
        return false;
    }

    /**
     * States if the block can hide the header.
     *
     * @return boolean success.
     */
    public function hide_header() {
        return false;
    }

    /**
     * Gets the content of the block.
     *
     * @return string Markup.
     */
    public function get_content() {
        if ($this->page->theme->name == 'boost') {
            return '';
        }
        if ($this->content !== null) {
            return $this->content;
        }

        $this->content = new stdClass();
        $this->content->text = core_course_drawer();

        return $this->content;
    }
}


version.php
<?php
// This file is part of Moodle - http://moodle.org/
//
// Moodle is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Moodle is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.

/**
 * Course Index block.
 *
 * @package    block_course_index
 * @copyright  &copy; 2022-onwards G J Barnard.
 * @author     G J Barnard - {@link http://moodle.org/user/profile.php?id=442195}
 * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
 */

defined('MOODLE_INTERNAL') || die;

$plugin->release = '4.0.0.2';
$plugin->version   = 2022011901;
$plugin->requires  = 2022011800.00; // 4.0dev+ (Build: 20220118).
$plugin->maturity = MATURITY_ALPHA;
$plugin->component = 'block_course_index';
